src/Event/CalendarConfigurationEvent.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Kimai time-tracking app.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Event;
+
+use Symfony\Contracts\EventDispatcher\Event;
+
+class CalendarConfigurationEvent extends Event
+{
+    /**
+     * @var array
+     */
+    private $configuration;
+
+    public function __construct(array $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
+    public function getConfiguration(): array
+    {
+        return $this->configuration;
+    }
+
+    public function setConfiguration(array $configuration)
+    {
+        $this->configuration = $configuration;
```

Comment:
Can you please use something like this:
```php
foreach ($configuration as $key => $value) {
  if (array_key_exists($key, $this->configuration) {
    $this->configuration[$key] = $value;
  }
}
```
Otherwise the plugins need to be clever enough to set all keys which might be added in the future.